specification/cognitiveservices/data-plane/Language/preview/2022-04-01-preview/textanalytics.json

```diff
@@ -2755,6 +3484,121 @@
       },
       "description": "if showStats=true was specified in the request this field will contain information about the document payload."
     },
+    "ConversationResult": {
+      "type": "object",
+      "required": [
+        "id",
+        "warnings"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Unique, non-empty conversation identifier."
+        },
+        "warnings": {
+          "type": "array",
+          "description": "Warnings encountered while processing document.",
+          "items": {
+            "$ref": "#/definitions/ConversationWarning"
+          }
+        },
+        "statistics": {
+          "$ref": "#/definitions/ConversationStatistics"
+        }
+      }
+    },
+    "ConversationError": {
+      "type": "object",
+      "required": [
+        "id",
+        "error"
+      ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The id of the conversation."
+        },
+        "error": {
+          "type": "object",
+          "description": "Conversation Error.",
+          "$ref": "common.json#/definitions/Error"
+        }
+      }
+    },
+    "ConversationWarning": {
+      "type": "object",
+      "required": [
+        "code",
+        "message"
+      ],
+      "properties": {
+        "code": {
+          "type": "string",
+          "enum": [
+            "ConversationTruncated"
+          ],
+          "x-ms-enum": {
+            "name": "WarningCodeValue",
+            "modelAsString": true
+          },
+          "description": "Error code."
+        },
+        "message": {
+          "type": "string",
+          "description": "Warning message."
+        },
+        "targetRef": {
+          "type": "string",
+          "description": "A JSON pointer reference indicating the target object."
```

Comment:
@annatisch @tg-msft any thoughts on how we might process this? Any prior art? In an SDK, a JSON path or pointer doesn't make sense in the public API, so I imagine we'd want to figure that out ourselves and use a pointer/reference ourselves.

Reply:
I don't know of JSON Pointer being used by any other service.  The only place I know of it in our ecosystem besides TextAnalytics is AutoRest README transforms.  We've had these in TA for a long time though and just parse by hand at the moment.  See https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs#L671.